third_party/move/move-compiler/src/parser/syntax.rs

```diff
@@ -553,20 +553,39 @@ impl Modifiers {
 // ModuleMemberModifiers checks for uniqueness, meaning each individual ModuleMemberModifier can
 // appear only once
 fn parse_module_member_modifiers(context: &mut Context) -> Result<Modifiers, Box<Diagnostic>> {
+    let check_previous_vis = |context: &mut Context, mods: &mut Modifiers, vis: &Visibility| {
+        if let Some(prev_vis) = &mods.visibility {
+            let msg = "Duplicate visibility modifier".to_string();
+            let prev_msg = "Visibility modifier previously given here".to_string();
+            context.env.add_diag(diag!(
+                Declarations::DuplicateItem,
+                (vis.loc().unwrap(), msg),
+                (prev_vis.loc().unwrap(), prev_msg),
+            ));
+        }
+    };
     let mut mods = Modifiers::empty();
     loop {
         match context.tokens.peek() {
             Tok::Public => {
                 let vis = parse_visibility(context)?;
```

Comment:
Update `parse_visibility` function's comment to include this new visibility syntax.